setup.py

```diff
@@ -247,17 +263,20 @@ def finalize_options(self):
                 print("#################################\n")
             # More portable to pass -fopenmp to linker.
             # self.libraries += ['gomp']
+            if self.compiler.compiler_type == 'msvc':
+                openmpflag = '/openmp'
+                compileflags = COMPILE_FLAGS_MSVC
+            else:
+                openmpflag = '-fopenmp'
+                compileflags = COMPILE_FLAGS
             for e in self.extensions:
-                if '-fopenmp' not in e.extra_compile_args:
-                    e.extra_compile_args += ['-fopenmp']
-                if '-fopenmp' not in e.extra_link_args:
-                    e.extra_link_args += ['-fopenmp']
+                if openmpflag not in e.extra_compile_args:
+                    e.extra_compile_args += [openmpflag]
+                    e.extra_compile_args += compileflags
```

Comment:
`compileflags` only gets added to `e.extra_compile_args` if compiling with openmp. Maybe move to the `_needs_stub` loop, but be careful that they only get added once (I seem to remember `build_extensions` being called multiple times being an issue.